core/utils/wasm-builder/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # WASM builder is an utility for building a project as a WASM binary
+//!
+//! The WASM builder is a tool that integrates the building of a WASM binary of your project into the main
+//! `cargo` build process.
+//!
+//! ## Project setup
+//!
+//! A project that should be compiled as WASM binary needs to add a `build.rs`, add
+//! `substrate-wasm-builder-runner` as dependency into `build-dependencies` and requires to have a
+//! feature called `no-std`. The `build.rs` needs to contain the following code:
+//!
+//! ```rust,nocompile
+//! use wasm_builder_runner::{build_current_project, WasmBuilderSource};
+//!
+//! fn main() {
+//! 	build_current_project(
+//! 		// The name of the file being generated in out-dir.
+//! 		"wasm_binary.rs",
+//! 		// How to include wasm-builder, in this case from crates.io.
+//! 		WasmBuilderSource::Crates("1.0.0"),
+//! 	);
+//! }
+//! ```
+//!
+//! The `no-std` feature will be enabled by WASM builder while compiling your project to WASM.
+//!
+//! As a last step you need to add the following to your project:
+//!
+//! ```rust,nocompile
+//! include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+//! ```
+//!
+//! This will include the generated wasm binary as two constants `WASM_BINARY` and `WASM_BINARY_BLOATY`.
```

Comment:
```suggestion
//! This will include the generated WASM binary as two constants `WASM_BINARY` and `WASM_BINARY_BLOATY`.
```

All `WASM` should probably be `Wasm`